server/src/main/java/io/crate/analyze/validator/SemanticSortValidator.java

```diff
@@ -88,19 +88,15 @@ public Void visitFunction(Function symbol, SortContext context) {
             if (!context.inFunction && !SUPPORTED_TYPES.contains(symbol.valueType().id())) {
                 throw new UnsupportedOperationException(
                     String.format(Locale.ENGLISH,
-                                  "Cannot %s '%s': invalid return type '%s'.",
-                                  context.operation,
-                                  symbol,
-                                  symbol.valueType())
+                        "Cannot %s '%s': invalid return type '%s'.",
+                        context.operation,
+                        symbol,
+                        symbol.valueType())
                 );
             }
-            try {
-                context.inFunction = true;
-                for (Symbol arg : symbol.arguments()) {
-                    arg.accept(this, context);
-                }
-            } finally {
-                context.inFunction = false;
+            context.inFunction = true;
+            for (Symbol arg : symbol.arguments()) {
+                arg.accept(this, context);
```

Comment:
since inFunction is not reset, do we check correctly in the `visitSymbol()`, other symbols in ORDER BY, example:
`SELECT * FROM t ORDER BY LEFT(txt_col, 1) = ANY(['a']), <array_col> `?

Reply:
(meaning that the data type of other plain symbols -not functions- are in the supported types?

Reply:
If I understood correctly, for `order by <exp1>, <exp2>`, exp1 and exp2 are validated with separate `SortContext` so the contexts are not shared.

Reply:
Cool, but please verify it, (maybe add a test if there is no such)